src/Sdk/DTPipelines/workflow-v1.0.json

```diff
@@ -162,18 +162,35 @@
         "snapshot-mapping"
       ]
     },
-
+    
     "snapshot-mapping": {
       "mapping": {
         "properties": {
           "image-name": {
             "type": "non-empty-string",
             "required": true
+          },
+          "if": "snapshot-if",
+          "version": {
+            "type": "non-empty-string",
+            "required": false
           }
         }
       }
     },
 
+    "snapshot-if": {
```

Comment:
do you need to have 
```
 "always(0,0)",
        "failure(0,0)",
        "cancelled(0,0)",
        "success(0,0)",
```
like other `if` expression?

Reply:
To start with no. After discussing this with @ericsciple, we want to start small and extend it later. So we chose only to include what we think is sufficient.